### PR TITLE
travis passing

### PR DESCRIPTION
**Pull request by Judahmeek**
- added yarn support
- updated dependencies (eslint-shakacode)
- lint passes
- specs passes
   - specs cover utils only though

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/bootstrap-loader/259)

### CODE REVIEW

**Comment by justin808** [comment]
@Judahmeek Thank you so much!

@alexfedoseev LGTM. OK with you?

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,9 @@
 language: node_js
 
 node_js:
-  - "6"
+  - "7"
```
Comment (alexfedoseev) [inline .travis.yml]: Let's test on at least 3 major versions?

```yaml
node_js:
  - "5"
  - "6"
  - "7"
```

File: src/utils/buildExtractStylesLoader.js [hunk 2]
```diff
@@ -35,5 +35,5 @@ Error: ${error}
       .map(loader => `${loader}!`)
       .join('')
   );
-  return ExtractTextPlugin.extract({ fallbackLoader, loader: restLoaders });
+  return ExtractTextPlugin.extract({ fallback: fallbackLoader, use: restLoaders });
```
Comment (alexfedoseev) [inline src/utils/buildExtractStylesLoader.js]: We have [this ongoing PR](https://github.com/shakacode/bootstrap-loader/pull/249). TBH I haven't looked into details of the issue and not sure why returned value must be a string (when it's changed to array in ETP). Can you check if this resolves https://github.com/shakacode/bootstrap-loader/issues/238 or not.

File: .travis.yml (hunk 1 above)
Comment (Judahmeek) [inline .travis.yml]: Sure thing.

File: src/utils/buildExtractStylesLoader.js (hunk 2 above)
Comment (Judahmeek) [inline src/utils/buildExtractStylesLoader.js]: This PR currently doesn't solve that problem. A string is required by our own code. I just created an amendment that will solve this problem and be completely backwards compatible.

File: examples/basic/package.json [hunk 3]
```diff
@@ -54,7 +54,7 @@
     "babel-loader": "^6.2.8",
     "babel-preset-es2015": "^6.18.0",
     "body-parser": "^1.15.2",
-    "bootstrap-loader": "file:../..",
+    "bootstrap-loader": "file:///Users/judahmeek/Apps/bootstrap-loader",
```
Comment (Judahmeek) [inline examples/basic/package.json]: https://github.com/shakacode/bootstrap-loader/blob/master/examples/basic/package.json#L58

File: src/utils/buildExtractStylesLoader.js [hunk 4]
```diff
@@ -35,5 +35,9 @@ Error: ${error}
       .map(loader => `${loader}!`)
       .join('')
   );
-  return ExtractTextPlugin.extract({ fallbackLoader, loader: restLoaders });
+  return [
+    `${ExtractTextPlugin.loader().loader}?{"omit":1,"remove":true}`,
+    fallbackLoader,
+    restLoaders,
+  ].join('!');
```
Comment (Judahmeek) [inline src/utils/buildExtractStylesLoader.js]: I'm fairly confident that this does not apply to webpack 1.  The latest commit for the extract-text-webpack-plugin supporting webpack 1 was back in June 15, 2016: https://github.com/webpack-contrib/extract-text-webpack-plugin/tree/webpack-1

**Pull request merged**